src/java/org/apache/cassandra/index/sai/disk/v1/MergeOneDimPointValues.java

```diff
@@ -91,32 +99,43 @@ public long getNumRows()
     @SuppressWarnings("resource")
     public void intersect(IntersectVisitor visitor) throws IOException
     {
-        while (queue.size() != 0)
+        try
         {
-            final BKDReader.IteratorState reader = queue.top();
-            if (reader.hasNext())
+            while (queue.size() != 0)
             {
-                final long rowID = reader.next();
-
-                minRowID = Math.min(minRowID, rowID);
-                maxRowID = Math.max(maxRowID, rowID);
-                numRows++;
-
-                visitor.visit(rowID, reader.scratch);
-
+                final BKDReader.IteratorState reader = queue.top();
                 if (reader.hasNext())
                 {
-                    queue.updateTop();
+                    final long rowID = reader.next();
+
+                    minRowID = Math.min(minRowID, rowID);
+                    maxRowID = Math.max(maxRowID, rowID);
+                    numRows++;
+
+                    visitor.visit(rowID, reader.scratch);
+
+                    if (reader.hasNext())
+                    {
+                        queue.updateTop();
+                    }
+                    else
+                    {
+                        queue.pop();
+                    }
                 }
                 else
                 {
-                    queue.pop();
+                    // iterator is exhausted
+                    BKDReader.IteratorState iterator = queue.pop();
+                    iterator.close();
```

Comment:
nit: Could be replaced with `queue.pop().close()`

Reply:
yup